src/ControlStream.c

```diff
@@ -66,6 +66,17 @@ typedef struct _QUEUED_ASYNC_CALLBACK {
             uint8_t g;
             uint8_t b;
         } setControllerLed;
+        struct {
+            uint16_t controllerNumber;
+            uint8_t type_left;
+            uint8_t type_right;
+            // arrays of size DS_EFFECT_PAYLOAD_SIZE
+            // this is an opaque payload that will be read directly from the joypad and set as is to the client controller
+            // if you are curious about the actual data, there's some rationale in
+            // https://gist.github.com/Nielk1/6d54cc2c00d2201ccb8c2720ad7538db
+            uint8_t *left;
+            uint8_t *right;
```

Comment:
Any reason not to just define the `left[DS_EFFECT_PAYLOAD_SIZE]` and `right[DS_EFFECT_PAYLOAD_SIZE]` arrays here and avoid having additional allocations?

Reply:
I was trying to keep the copies at a minimum, and I started with a pointer in the callback function definition. I agree though, better to keep the code simple and avoid another mallov; I've updated the code to use plain arrays instead.